api.go

```diff
@@ -0,0 +1,2006 @@
+
+// String returns the string representation
+func (s EcsIndexableKey) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s EcsIndexableKey) GoString() string {
+	return s.String()
+}
+
+// SetDatatype sets the Datatype field's value.
+func (s *EcsIndexableKey) SetDatatype(v string) *EcsIndexableKey {
+	s.Datatype = &v
+	return s
+}
+
+// SetName sets the Name field's value.
+func (s *EcsIndexableKey) SetName(v string) *EcsIndexableKey {
+	s.Name = &v
+	return s
+}
+
+type EcsObjectMatch struct {
+	_ struct{} `type:"structure"`
+
+	IndexKey        *string              `locationName:"indexKey" type:"string"`
+	ObjectId        *string              `locationName:"objectId" type:"string"`
+	ObjectName      *string              `locationName:"objectName" type:"string"`
+	ObjectOwnerZone *string              `locationName:"objectOwnerZone" type:"string"`
+	QueryMds        []*EcsQueryMds `locationName:"queryMds" type:"list" flattened:"true"`
+	VersionId       *string              `locationName:"versionId" type:"string"`
+}
+
+// String returns the string representation
+func (s EcsObjectMatch) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s EcsObjectMatch) GoString() string {
+	return s.String()
+}
+
+// SetIndexKey sets the IndexKey field's value.
+func (s *EcsObjectMatch) SetIndexKey(v string) *EcsObjectMatch {
+	s.IndexKey = &v
+	return s
+}
+
+// SetObjectId sets the ObjectId field's value.
+func (s *EcsObjectMatch) SetObjectId(v string) *EcsObjectMatch {
+	s.ObjectId = &v
+	return s
+}
+
+// SetObjectName sets the ObjectName field's value.
+func (s *EcsObjectMatch) SetObjectName(v string) *EcsObjectMatch {
+	s.ObjectName = &v
+	return s
+}
+
+// SetObjectOwnerZone sets the ObjectOwnerZone field's value.
+func (s *EcsObjectMatch) SetObjectOwnerZone(v string) *EcsObjectMatch {
+	s.ObjectOwnerZone = &v
+	return s
+}
+
+// SetQueryMds sets the QueryMds field's value.
+func (s *EcsObjectMatch) SetQueryMds(v []*EcsQueryMds) *EcsObjectMatch {
+	s.QueryMds = v
+	return s
+}
+
+// SetVersionId sets the VersionId field's value.
+func (s *EcsObjectMatch) SetVersionId(v string) *EcsObjectMatch {
+	s.VersionId = &v
+	return s
+}
+
+type EcsOptionalAttribute struct {
+	_ struct{} `type:"structure"`
+
+	Datatype *string `type:"string"`
+	Name     *string `type:"string"`
+}
+
+// String returns the string representation
+func (s EcsOptionalAttribute) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s EcsOptionalAttribute) GoString() string {
+	return s.String()
+}
+
+// SetDatatype sets the Datatype field's value.
+func (s *EcsOptionalAttribute) SetDatatype(v string) *EcsOptionalAttribute {
+	s.Datatype = &v
+	return s
+}
+
+// SetName sets the Name field's value.
+func (s *EcsOptionalAttribute) SetName(v string) *EcsOptionalAttribute {
+	s.Name = &v
+	return s
+}
+
+type EcsQueryMds struct {
```

Comment:
after discussion with David on slack. the change will be "EcsQueryMds -> QueryMetadata (or EcsQueryMetada) and MdType -> MetadataType (or EcsMetadataType), and leave the other two."